src/main/kotlin/org/rust/ide/intentions/visibility/ChangeVisibilityIntention.kt

```diff
@@ -43,10 +43,15 @@ abstract class ChangeVisibilityIntention : RsElementBaseIntentionAction<ChangeVi
     }
 
     companion object {
-        fun isValidVisibilityOwner(visible: RsVisibilityOwner): Boolean =
-            visible !is RsEnumVariant
-                && visible.parent.parent !is RsTraitItem
-                && visible.containingCrate?.origin == PackageOrigin.WORKSPACE
+        fun isValidVisibilityOwner(visible: RsVisibilityOwner): Boolean {
+            if (visible is RsEnumVariant) return false
+            if (visible.containingCrate?.origin != PackageOrigin.WORKSPACE) return false
+
+            val traitOrImpl = visible.parent.parent as? RsTraitOrImpl
```

Comment:
```suggestion
            val traitOrImpl = visible.context?.context as? RsTraitOrImpl
```
Otherwise, this doesn't work with members expanded from macros (macro calls inside trait impls)